### DISCUSSION TITLE
MySQL 8.0.27 not connecting to Budibase 1.0.26 (Ecconrefused 3306 Status 500)

### DISCUSSION THREAD

**Discussion by HoffmanM361** [Budibase/budibase#3999] [Help]
MySQL 8.0.27 is not connecting to Budibase 1.0.26

Note: We connect to "MySQL workbench 8.0.27" with a variety of user names with and without SSL
Host: localhost
Port: 3306
User names tested: root, phruser
Database Name: phr

Errors include Ecconrefused and Etimedout.  Lately, Ecconrefused

If you don't mind, where are the location of the error logs, (the file in the directory we made "budi" we  is empty), or how are error logs turned on? 

Is there a problem with the version, is MySQL 8 supported? 

Ubuntu Focal Desktop.  Firefox Browser. 

Any help is appreciated:)

**Comment by mike12345567** [comment]
Hi @HoffmanM361 - you've setup your host as `localhost` - however if you're running in self host with Docker then localhost will reference the container in which Budibase is running, rather than the actual host system that the database is running in (or another container if it is also running in Docker).

To access the host machine on Windows or MacOS then you can use `host.docker.internal` as the host, for Linux use the IP address `172.17.0.1` which is attached to the host.

You can find discussion around this: https://stackoverflow.com/questions/31324981/how-to-access-host-port-from-docker-container

Currently error logs are not stored anywhere on the host machine, the best way to see logs of what is happening is looking at the docker logs:
1. Run `docker container ls` to list the running Docker containers
2. Run `docker logs <container id>` using the ID of the server container (the Budibase server carries out all work for apps, the worker carries out global operations pertaining to the whole Budibase cluster e.g. user ops).

**Comment by slickers** [comment]
I have successfully connect BB to host MYSQL with these step:

1. set IP to 172.17.0.1 (linux machine)
2. create db user = username@%  --> % is replacement of ANY, use PHPMYADMIN 
3. allow remote access to mysql (search : How To Allow Remote Access to MySQL)
--> sudo nano /etc/mysql/mysql.conf.d/mysqld.cnf
--> bind-address = 127.0.0.1 --> bind-address = [ip.of.mach.ine] or 0.0.0.0
--> sudo systemctl restart mysql
